builtin/common/serialize.lua

```diff
@@ -195,7 +198,7 @@ function core.deserialize(str, safe)
 	end
 	local f, err = loadstring(str)
 	if not f then return nil, err end
-	setfenv(f, safe and safe_env or env)
+	setfenv(f, safe and {loadstring = function() end} or env)
```

Comment:
Actually, it's better to move this back into a `local safe_env = {` above so the function is not unnecessarily recreated on each call.

Reply:
Added.

Reply:
`env` needs to be newly constructed on every invocation.